instant-distance-py/src/lib.rs

```diff
@@ -24,41 +24,79 @@ fn instant_distance_py(_: Python, m: &PyModule) -> PyResult<()> {
     m.add_class::<Search>()?;
     m.add_class::<Hnsw>()?;
     m.add_class::<HnswMap>()?;
+    m.add_class::<DistanceMetric>()?;
     Ok(())
 }
 
+#[pyclass]
+#[derive(Copy, Clone)]
+enum DistanceMetric {
+    Euclid,
+    Cosine,
```

Comment:
Given that we don't actually have a `Cosine` implementation at this point, we should not introduce this variant here until the commit where we add the backing implementation.